src/components/sandpack-embed/index.tsx

```diff
@@ -1,44 +1,77 @@
 import { Box, BoxProps } from '@chakra-ui/react'
-import { Sandpack, SandpackProps } from '@codesandbox/sandpack-react'
-import '@codesandbox/sandpack-react/dist/index.css'
+import {
+  CodeEditorProps,
+  PreviewProps,
+  SandpackCodeEditor,
+  SandpackLayout,
+  SandpackLayoutProps,
+  SandpackPreview,
+  SandpackProvider,
+} from '@codesandbox/sandpack-react'
+import { nightOwl } from '@codesandbox/sandpack-themes'
 
-type Props = BoxProps &
-  SandpackProps & {
-    dependencies?: Record<string, string>
-    devDependencies?: Record<string, string>
+type Props = BoxProps & {
+  dependencies?: Record<string, string>
+  devDependencies?: Record<string, string>
+  layoutOptions?: BoxProps & SandpackLayoutProps
+  editorOptions?: CodeEditorProps
+  previewOptions?: PreviewProps
+  files: {
+    [x: string]: string
   }
+}
 
-const SandpackEmbed = ({ dependencies, devDependencies, ...props }: Props) => (
-  <Box
-    as={Sandpack}
-    {...props}
-    options={{
-      ...props.options,
-      showLineNumbers: true,
-    }}
-    theme='dark'
-    template='react-ts'
-    customSetup={{
-      dependencies: {
-        'react-icons': '3.11.0',
-        '@chakra-ui/react': 'latest',
-        '@chakra-ui/icons': 'latest',
-        '@emotion/react': '^11.7.0',
-        '@emotion/styled': '^11.6.0',
-        'framer-motion': '^4.1.17',
-        react: '^18.0.0',
-        'react-dom': '^18.0.0',
-        'react-scripts': '^4.0.0',
-        ...dependencies,
-      },
-      devDependencies: {
-        '@types/react': '^18.0.0',
-        '@types/react-dom': '^18.0.0',
-        typescript: '^4.0.0',
-        ...devDependencies,
-      },
-    }}
-  />
-)
+const SandpackEmbed = ({
+  dependencies,
+  devDependencies,
+  layoutOptions,
+  editorOptions,
+  previewOptions,
+  files,
+}: Props) => {
+  return (
+    <SandpackProvider
+      files={files}
+      theme={nightOwl}
+      template='react-ts'
+      customSetup={{
+        dependencies: {
+          'react-icons': '3.11.0',
+          '@chakra-ui/react': 'latest',
+          '@chakra-ui/icons': 'latest',
+          '@emotion/react': '^11.7.0',
+          '@emotion/styled': '^11.6.0',
+          'framer-motion': '^4.1.17',
+          react: '^18.0.0',
+          'react-dom': '^18.0.0',
+          'react-scripts': '^4.0.0',
+          ...dependencies,
+        },
+        devDependencies: {
+          '@types/react': '^18.0.0',
+          '@types/react-dom': '^18.0.0',
+          typescript: '^4.0.0',
+          ...devDependencies,
+        },
+      }}
+    >
+      <Box
+        as={SandpackLayout}
+        sx={{ '& > *': { height: '600px !important' } }}
+        {...layoutOptions}
+      >
+        <SandpackCodeEditor
+          showLineNumbers
+          style={{ flexBasis: '20%' }}
+          {...editorOptions}
+        />
+        <SandpackPreview
+          style={{ width: '100%', zIndex: 0 }}
+          {...previewOptions}
+        />
+      </Box>
```

Comment:
@noobinthisgame This could definitely be doable for the doc pages.

This is what it looks like on my external display currently (1920 x 955 screen size):
![image](https://user-images.githubusercontent.com/65234762/175318314-114fbcd5-d227-40d0-82c1-c31b23bf1380.png)

And after if simply forcing `flexDirection: column` and adjust the preview height:
![image](https://user-images.githubusercontent.com/65234762/175314261-1f64b570-8f91-4d56-80b0-7fd80e65ca12.png)

If we were to retain the horizontal stacking, we would have to expand the width of the overall main content container, which will at least effect the `TableOfContent` component (I guess it could be hidden at a larger breakpoint to prevent unintended layout compression? 🤷🏼‍♂️ )

I do not think this is needed for the embed that sits on the home page, because it already takes up a lot of real estate. 

We could force the wrap at a larger breakpoint to help with view on the tablet. (`min-width` for the editor and preview components are at `350px` and can bump it to `450px` for the editor) But we have a ancestor container changing to a smaller max width at the 768px breakpoint.

Reply:
The second picture is what I had in mind except that the preview should be on top of the code, as in all our code examples. Otherwise this looks great !

Reply:
Right!

How about this?
![image](https://user-images.githubusercontent.com/65234762/175793114-980962c9-b933-4354-9a45-2b911b12e2b3.png)